scipy/stats/_distribution_infrastructure.py

```diff
@@ -2182,8 +2261,8 @@ def _logentropy_logexp_safe(self, **params):
 
     def _logentropy_quadrature(self, **params):
         def logintegrand(x, **params):
-            logpdf = self._logpdf_dispatch(x, **params)
-            return logpdf + np.log(0j+logpdf)
+            logpxf = self._logpxf_dispatch(x, **params)
```

Comment:
Part of change 1: referring to "PXF" instead of "PDF"/"PMF" does the right thing for both continuous and discrete distributions. Note that `self._quadrature` below uses `tanhsinh` for continuous and `nsum` for discrete.